lasagne/objectives.py

```diff
@@ -41,11 +56,21 @@ def multinomial_nll(x, t):
             a 1D integer array that gives the class index of each sample
             (the position of the 1 in the row in a 1-of-N encoding, or
             1-hot encoding),
+        - m : mask; None for no mask, or a 1-dimensional array, that
+            selects which rows in `x` and `t` should count towards the
+            result
 
     :returns:
         - output : the mean multinomial negative log loss
     """
-    return T.mean(T.nnet.categorical_crossentropy(x, t))
+    if m is None:
+        return -T.mean(T.log(x)[T.arange(t.shape[0]), t])
```

Comment:
That's not the same as before. `t` can explicitly be either an integer array (the case you assumed here) or a 2D tensor of targets in one-hot encoding. If it's too difficult to treat both in one function, we should think about explicitly splitting it up.